core/services/pipeline/common.go

```diff
@@ -237,6 +237,11 @@ const (
 	TaskTypePanic TaskType = "panic"
 )
 
+var (
+	stringType = reflect.TypeOf("")
+	int32Type  = reflect.TypeOf(int32(0))
+)
+
 func UnmarshalTaskFromMap(taskType TaskType, taskMap interface{}, dotID string, config Config, txdb *gorm.DB, txdbMutex *sync.Mutex, nPreds int) (_ Task, err error) {
```

Comment:
Would be great to add some early assertions here - I think we could check that all the $(dotID.X) actually reference real dotIDs in the graph and fail fast if not

Reply:
Yeah @archseer was thinking of doing a followup PR that adds a bunch of validation, including doing some type checking and stuff like what you're suggesting